marimo/_runtime/runtime.py

```diff
@@ -959,6 +960,9 @@ def _delete_variables(
                     )
             else:
                 if name in self.globals:
+                    if isinstance(self.globals[name], threading.Thread):
+                        # If the variable is a thread, we need to stop it
+                        self.globals[name].join()
```

Comment:
I don't think this is right. This will block the kernel as long as the thread is running. I'd rather leak the thread, or find a way to terminate it without rendering the kernel unusable.

You can try by running the following in a `mo.Thread` then deleting the cell:

```python
def foo():
  import time
  while True: time.sleep(1)
```

Reply:
It's not right, and with the current smoke test won't even trigger since it needs a bound name to even get here.